browser/psst/brave_psst_permission_context_factory.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/psst/brave_psst_permission_context_factory.h"
+
+#include "brave/components/psst/browser/core/brave_psst_permission_context.h"
+#include "brave/components/psst/common/features.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+
+namespace psst {
+
+// static
+BravePsstPermissionContextFactory*
+BravePsstPermissionContextFactory::GetInstance() {
+  static base::NoDestructor<BravePsstPermissionContextFactory> instance;
+  return instance.get();
+}
+
+// static
+BravePsstPermissionContext*
+BravePsstPermissionContextFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  CHECK(context);
+  CHECK(base::FeatureList::IsEnabled(psst::features::kEnablePsst));
+  return static_cast<BravePsstPermissionContext*>(
+      GetInstance()->GetServiceForContext(context, true));
+}
+
+BravePsstPermissionContextFactory::BravePsstPermissionContextFactory()
+    : ProfileKeyedServiceFactory("BravePsstPermissionService") {
+  DependsOn(HostContentSettingsMapFactory::GetInstance());
+}
+
+BravePsstPermissionContextFactory::~BravePsstPermissionContextFactory() =
+    default;
+
+std::unique_ptr<KeyedService>
+BravePsstPermissionContextFactory::BuildServiceInstanceForBrowserContext(
+    content::BrowserContext* context) const {
+  CHECK(context);
+  auto* profile = Profile::FromBrowserContext(context);
+  auto* host_content_settings_map =
+      HostContentSettingsMapFactory::GetForProfile(profile);
+
+  CHECK(host_content_settings_map);
```

Comment:
I think this really isn't necessary, if there's a CHECK anywhere it should be inside `BravePsstPermissionContext`